lib/VM/JSLib/RegExp.cpp

```diff
@@ -595,6 +658,13 @@ CallResult<Handle<JSArray>> directRegExpExec(
   auto inputSHV = SmallHermesValue::encodeStringValue(*S, runtime);
   JSObject::setNamedSlotValueUnsafe(*A, runtime, inputDesc, inputSHV);
 
+  // Create indices array.
+  auto indicesRes = JSArray::create(runtime, match.size(), match.size());
```

Comment:
Why do you create this unconditionally? This will only ever be used if `hasIndices` is true, right?

Reply:
I created at this point because I need this array already declared in order to be used inside the loop [here](https://github.com/fabioh8010/hermes/blob/feature/regexp-indices-flag/lib/VM/JSLib/RegExp.cpp#L691), and would also need the same array outside the loop [here](https://github.com/fabioh8010/hermes/blob/feature/regexp-indices-flag/lib/VM/JSLib/RegExp.cpp#L724). Do you think it's problematic?